doc/replication.md

```diff
@@ -68,65 +66,62 @@ devices. This web service needs to:
 ### Replication on the Device
 
 From the device side, replication is straightforward. You can replicate from a
-local datastore to a remote database, from a remote database to a local
-datastore, or both ways to implement synchronisation.
+local document store to a remote database, from a remote database to a local
+document store, or both ways to implement synchronisation.
 
 Replications are set up in code on a device. Use `ReplicatorBuilder` to create
 a pre-configured `Replicator` object. Each `Replicator`
 object can be assigned a `Listener` to receive message when replication
 completes or encounters an error.
 
+The replication examples below are 'one-shot' and will run once until
+completion, at which point the local and remote databases will be
+synchronised. [Replication policies](replication-policies.md) allow
+for more complex "policies" to be executed, eg "only replicate when on
+Wifi" or "replicate every 30 minutes".
+
 First we create a simple listener that just sets a CountDownLatch when the
 replication finishes so we can wait for a replication to finish without
 needing to poll:
 
 ```java
-import com.cloudant.sync.event.Subscribe;
-import java.util.concurrent.CountDownLatch;
-import com.cloudant.sync.event.notifications.ReplicationCompleted;
-import com.cloudant.sync.event.notifications.ReplicationErrored;
-import com.cloudant.sync.replication.ErrorInfo;
-
 /**
  * A {@code ReplicationListener} that sets a latch when it's told the
  * replication has finished.
  */
 private class Listener {
 
     private final CountDownLatch latch;
-    public ErrorInfo error = null;
-    public int documentsReplicated;
-    public int batchesReplicated;
+    public List<Throwable> errors = new ArrayList<Throwable>();
+    public int documentsReplicated = 0;
+    public int batchesReplicated = 0;
 
     Listener(CountDownLatch latch) {
         this.latch = latch;
     }
 
     @Subscribe
     public void complete(ReplicationCompleted event) {
-        this.documentsReplicated = event.documentsReplicated;
-        this.batchesReplicated = event.batchesReplicated;
+        this.documentsReplicated += event.documentsReplicated;
+        this.batchesReplicated += event.batchesReplicated;
         latch.countDown();
     }
 
     @Subscribe
     public void error(ReplicationErrored event) {
-        this.error = event.errorInfo;
+        this.errors.add(event.errorInfo);
         latch.countDown();
     }
 }
 ```
 
-Next we replicate a local datastore to a remote database:
+Next we replicate a local document store to a remote database:
 
 ```java
-import com.cloudant.sync.replication.ReplicatorBuilder;
-import com.cloudant.sync.replication.Replicator;
-
 // Username/password are supplied in the URL and can be Cloudant API keys
 URI uri = new URI("https://username:password@username.cloudant.com/my_database");
```

Comment:
I'm not sure if "username" is a known account, but this should probably be updated to "examples.cloudant.com" or something in the IANA documentation namespace (although `cloudant.com` probably provides a more clear example than `cloudant.example`).